### PR TITLE
Support for HDR Output + HDR Apps

### PR DESCRIPTION
**Pull request by Joshua-Ashton**
Hooks up HDR support in the Gamescope WSI layer and also on the DRM side.

Can also output HDR clients -> SDR for debugging (where the "tonemapping" is just clipping).

There are a lot of cool things we can do for HDR on SDR and SDR on HDR, but this is good enough for an initial implementation :)

For correctness if you want to try it out, you'll need to use either my branch https://gitlab.freedesktop.org/JoshuaAshton/linux-hdr or hwentland's branch for correct colors.

You also need to use the Gamescope WSI layer. Eg

`ENABLE_GAMESCOPE_WSI=1 gamescope -w 3840 -h 2160 -W 3840 -H 2160 --hdr-enabled -- /home/joshua/Code/Granite/build-glfw/tests/hdr-test`

### CODE REVIEW

**Comment by Joshua-Ashton** [comment]
Going to merge this for now, as things seem to be working. It's going to be a huge pain to rebase stuff ontop of this as there's a lot.

File: src/drm.cpp [hunk 1]
```diff
@@ -1805,6 +1840,16 @@ int drm_prepare( struct drm_t *drm, bool async, const struct FrameInfo_t *frameI
 		if (ret < 0)
 			return ret;
 
+		drm->connector->pending.colorspace = g_bOutputHDREnabled ? DRM_MODE_COLORIMETRY_BT2020_RGB : DRM_MODE_COLORIMETRY_DEFAULT;
```
Comment (swick) [inline src/drm.cpp]: Does `DRM_MODE_COLORIMETRY_BT2020_RGB` work for you? How about `DRM_MODE_COLORIMETRY_BT2020_YCC`? IIRC amdgpu will default to YCC when it's available.

Comment (Joshua-Ashton) [reply src/drm.cpp]: If I am honest, I have not seen any output difference with the DP colorimetry packet. Every monitor I have encountered seems to assume BT2020 whenever the PQ EOTF is selected.

It's on my to-do to investigate it further. I'll make a tracking issue.

Comment (Joshua-Ashton) [reply src/drm.cpp]: https://github.com/Plagman/gamescope/issues/732

**Comment by colemickens** [comment]
Beyond just refreshing this page: https://gitlab.freedesktop.org/JoshuaAshton/linux-hdr/-/commits/josh-hdr-colorimetry is there somewhere I can follow this AMD + HDR work?  (I'm not trying to sign people up for more work but a arewehdr.org would be neat. :smile: )

**Comment by Joshua-Ashton** [comment]
I think Harry Wentland is working on upstreaming the color space bits.

Once that's all in place we can look at getting stuff working for users in Big Picture mode by having Gamescope launch the Deck UI and DRM lease a display.

**Comment by MrDuartePT** [comment]
One question this will work on Nvidia i have Legion 5 laptop with RTX 3070 and gamescope work amazing well i should say never have problems, even using the MUX swith (igpu disable)

**Pull request merged**